File: Java/Niveau1/7- Conditions avancées, opérateurs booléens/Zones de couleurs.java
import algorea.Scanner;

class Main{
    static Scanner entrée = new Scanner(System.in);     

    public static void main(String[] args){
        int nombreCoordonnées = entrée.nextInt();
        for(int i = 1; i <= nombreCoordonnées; i++){
            int xJeton = entrée.nextInt();
            int yJeton = entrée.nextInt();
            boolean dansFeuille = (((xJeton > 0) && (xJeton < 90)) && ((yJeton > 0) && (yJeton < 70)));
            boolean dansZoneJauneMilieu = (((xJeton > 25) && (xJeton < 50)) && ((yJeton > 20) && (yJeton < 45)));
            if(!dansFeuille){
                print("En dehors de la feuille");
            }else if((yJeton < 60) && (((xJeton < 15) && (xJeton < 45)) || ((xJeton > 60) && (xJeton < 85)))){
                print("Dans une zone rouge");
            }else if((!dansZoneJauneMilieu) && ((xJeton > 10) && (xJeton < 85)) && ((yJeton > 10) && (yJeton < 55))){
                print("Dans une zone bleue");
            }else{
                print("Dans une zone jaune")
            }
        }
        
    }

    private static void print(Object object){
        System.out.println(object);
    }
}

File: C++/Niveau 1/7 - Conditions avancées, opérateurs booléens/Zones de couleurs.cpp
#include <iostream>
using namespace std;

int main(){

    int nombreJeton;

    cin >> nombreJeton;

    for(int i = 1; i <= nombreJeton; i++){

        int x, y;

        cin >> x >> y;

        bool estDansLaFeuille = (((0 <= x) && (x <= 90)) && ((0 <= y) && (y <= 70)));

        bool estDaneZone1Rouge = (((60 <= y) && (y <= 70)) && ((15 <= x) && (x <= 45)));
        bool estDaneZone2Rouge = (((60 <= y) && (y <= 70)) && ((60 <= x) && (x <= 85)));
        bool estDansZoneRouge = (estDaneZone1Rouge ||estDaneZone2Rouge);

        bool estDaneZone1Bleue = (((10 <= y) && (y <= 20)) && ((10 <= x) && (x <= 85)));
        bool estDaneZone2Bleue = (((20 <= y) && (y <= 45)) && ((10 <= x) && (x <= 25)));
        bool estDaneZone3Bleue = (((20 <= y) && (y <= 45)) && ((50 <= x) && (x <= 85)));
        bool estDaneZone4Bleue = (((45 <= y) && (y <= 55)) && ((10 <= x) && (x <= 85)));
        bool estDansZoneBleue = (estDaneZone1Bleue || estDaneZone2Bleue || estDaneZone3Bleue || estDaneZone4Bleue);

        if(!estDansLaFeuille){
            cout << "En dehors de la feuille" << endl;
        }else if(estDansZoneRouge){
            cout << "Dans une zone rouge" << endl;
        }else if(estDansZoneBleue){ 
            cout << "Dans une zone bleue" << endl;
        }else{
            cout << "Dans une zone jaune" << endl;
        }

    }


}   